### PR TITLE
Fix parsing for `max_determinized_states`

### LINKED ISSUE #22722: max_determinized_states in query_string is no longer supported
We're upgrading client to elastic 5.1.2 and we noticed that the `max_determinized_states` is no longer accepted by the `query_string` query.

The following query is accepted by elastic 2.x:
```
curl localhost:9200/twitter/_search -d '{"query":{"query_string":{"query":"test", "max_determinized_states":10}}}'
```

But will fail on elastic 5.1.2 with:
```
"type":"parsing_exception","reason":"[query_string] query does not support [max_determinized_states]"
```

I don't think it's intentional because this option is still referenced in the documentation (and actually very useful).

### PR DESCRIPTION
**Pull request by nik9000**
There was a typo in the `ParseField` declaration. I know
we want to port these parsers to `ObjectParser` eventually
but I don't have the energy for that today and want to get
this fixed.

Closes #22722

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java [hunk 1]
```diff
@@ -92,7 +91,7 @@
     private static final ParseField QUOTE_ANALYZER_FIELD = new ParseField("quote_analyzer");
     private static final ParseField ALLOW_LEADING_WILDCARD_FIELD = new ParseField("allow_leading_wildcard");
     private static final ParseField AUTO_GENERATE_PHRASE_QUERIES_FIELD = new ParseField("auto_generate_phrase_queries");
-    private static final ParseField MAX_DETERMINED_STATES_FIELD = new ParseField("max_determined_states");
+    private static final ParseField MAX_DETERMINED_STATES_FIELD = new ParseField("max_determinized_states");
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java]: When I backport I'll add `max_determined_states` as a deprecated name.

**Review by nik9000 — COMMENTED** [review]
I'll add a breaking changes not for this in a moment.

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
> LGTM

Thanks!

**Comment by nik9000** [comment]
master: ee264c69573dd5b68f5e8f4649f6964c6582aed6
5.x:

File: core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java (hunk 1 above)
Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java]: shouldn't we have changed the name of the constant too?

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java]: Well caught! I'll change it.

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java]: I pushed 2e399e5505797cf5916208eefb13338a5d6f66c9

**Pull request merged**